cpp/benchmarks/sort/rank_lists.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "nested_types_common.hpp"
+
+#include <cudf/sorting.hpp>
+
+#include <nvbench/nvbench.cuh>
+
+void nvbench_rank_lists(nvbench::state& state, cudf::rank_method method)
```

Comment:
Should `rank_method` be an axis, instead of creating separate benchmarks for every option?

Reply:
It could be, but that means we will be compiling that many more templates. I figured that we have 4 possible options, and we shouldn't re-compile the function-template for every option

Reply:
See suggestion below.